packages/components/src/components/Snackbar/Snackbar.tsx

```diff
@@ -42,20 +66,231 @@ type SnackbarProviderProps = {
+      // bottom: 40,
+      // flexGrow: 1,
+      flexDirection: 'row',
+      flexWrap: 'wrap',
+      gap: 5,
+      minHeight: 44,
+      padding: sizing._12,
+      marginHorizontal: sizing._20,
+    },
+  }
+
+  const iconAndMessageContainer: ViewProps = {
+    accessible: true,
+    role: 'alert',
+    style: {
+      flexDirection: 'row',
+      flexGrow: 1,
+    },
+  }
+
+  const iconViewStyle: ViewStyle = {
+    // Below keeps icon aligned with first row of text, centered, and scalable
+    alignSelf: 'flex-start',
+    minHeight: 22 * fontScale,
+    alignItems: 'center',
+    justifyContent: 'center',
+  }
+
+  const iconProps: IconProps = {
+    name: isError ? 'Warning' : 'CheckCircle',
+    fill: contentColor,
+    height: sizing._16,
+    width: sizing._16,
+    preventScaling: true,
+  }
+
+  const messageProps: TextProps = {
+    'aria-label': messageA11y,
+    style: {
+      color: contentColor,
+      flexShrink: 1,
+      ...typography.helperText,
+    },
+  }
+
+  const buttonContainer: ViewProps = {
+    style: {
+      // alignContent: 'flex-end',
+      flex: 0,
+      flexDirection: 'row',
+      flexWrap: 'wrap',
+      marginLeft: 'auto', // Maintains alignment to right side
+      paddingLeft: sizing._16, // Minimum spacing to message text
+      // alignSelf: 'flex-end',
+      // justifyContent: 'flex-end'
+    },
+  }
+
+  const actionButton = () => {
+    if (!onActionPressed) return null
+
+    const onPress = () => {
+      onActionPressed()
+      toast.onHide()
+    }
+    const actionText = isError ? t('tryAgain') : t('undo')
+
+    return (
+      <>
+        <SnackbarButton text={actionText} onPress={onPress} />
+        <Spacer size={sizing._24} horizontal />
+      </>
+    )
+  }
+
+  return (
+    <View {...containerProps}>
+      <View {...iconAndMessageContainer}>
+        <View style={iconViewStyle}>
+          <Icon {...iconProps} />
+        </View>
+        <Spacer size={sizing._8} horizontal />
+        <Text {...messageProps}>{toast.message}</Text>
+      </View>
+      <View {...buttonContainer}>
+        {actionButton()}
+        <SnackbarButton text={t('dismiss')} onPress={toast.onHide} />
+      </View>
+    </View>
+  )
+}
+
+/**
+ * Convenience handling function to show snackbar
+ * @param message - message text to display on snackbar
+ * @param snackbarOptions - data to customize snackbar behavior for more than standard dismissible success
+ */
+export const ShowSnackbar = (
+  message: string,
+  snackbarOptions?: snackbarData['data'],
+) => {
+  snackbar.hideAll() // Remove any existing snackbars
+  snackbar.show(message, { data: snackbarOptions })
+}
+
+/**
+ * Convenience handling function to close snackbar(s)
+ */
+export const CloseSnackbar = () => {
+  snackbar.hideAll() // Remove any existing snackbars
+}
+
+/**
+ * Convenience handling function to close snackbar(s) on navigation
+ */
+export const CloseSnackbarOnNavigation = () => {
+  CloseSnackbar()
+}
```

Comment:
Not sure about these, especially the latter two. During tech discovery I thought they'd add something, but I'm not sure they do.

At some point I thought they'd be useful for analytics, but I don't think the Snackbar actually needs analytics. If we add an analytics passthrough to these, an app may as well just make their own versions and do the trivial `hideAll()` and `show()` themselves. The action button takes custom onPress logic that they can bake analytics into. Analytics on the "Dismiss" button doesn't really make sense--presumably every Snackbar that appears is dismissed, either by the user or switching screens. I also thought maybe there'd be some nice abstraction to mitigate breaking changes using these vs direct calls, but as things evolved I don't think there's anything that typing on `snackbar` can't handle.

Maybe we keep them (or just ShowSnackbar) on principle solely for slight convenience to a future app that doesn't care to have analytics on Snackbars to have a function call vs littering individual `snackbar` calls all over?

Reply:
Agreed that the third seems needed as it's the same as `CloseSnackbar()`.

I could see both Show and Close being useful as users might be expecting `CloseSnackbar` with the existence of `ShowSnackbar`.

As an aside, is exposing `snackbar` globally required for this to function? If not, maybe it would make more sense for users to only expose the `ShowSnackbar` and `CloseSnackbar` functions which they import.

Reply:
> Agreed that the third seems unneeded as it's the same as `CloseSnackbar()`.

Yeah, I mainly separated it as it matches flagship's having this function and calling attention to close on navigation as we can't inherently force apps to close Snackbars on navigation.

> As an aside, is exposing `snackbar` globally required for this to function?

Yes. Without the global var `snackbar` set, the `show` and `hideAll` functions aren't set to anything to control behavior.

We could conceivably package the component up differently--I think flagship [implemented it globally](https://www.npmjs.com/package/react-native-toast-notifications?activeTab=readme#--how-to-call-toast-outside-react-components) so that it could be tied into redux to update the bottom offset although I'm not that's even actually needed. The [standard implementation](https://www.npmjs.com/package/react-native-toast-notifications?activeTab=readme#usage) of the library uses a global wrapping component + hook in every place a Toast is called so it can only be called via react-native components. Seemed prudent to minimize the invasiveness for flagship and generally have a more powerful/robust implementation for future apps to maintain the global availability.

The global var `snackbar` does the heavy lifting, these functions are purely for convenience. `ShowSnackbar` actually does something in it enforcing closing any existing Snackbars first, but the `Close-` versions are just wrappers for `snackbar.hideAll()` that a properly configured app should be able to call anywhere.

Reply:
I do like the standard implementation of having the `useToast` hook. If we can confirm that the app wouldn't need to use the global for redux, I feel like just having either a `useSnackbar` hook OR importing `ShowSnackbar` and `CloseSnackbar` is cleaner than giving them the option to use the global `snackbar` too.

Reply:
@narin and I discussed in Slack and we have added a note to ticket #415 to look into shifting to use the [standard implementation](https://www.npmjs.com/package/react-native-toast-notifications?activeTab=readme#usage) of the toast library which may simplify some things for other follow-up tickets. The main con being it complicating the implementation for flagship, but this would be a one-time downside for otherwise more standard react(-native) functional implementation of the Snackbar component.

If everything pans out, global implementation can be explored post-MVP if/when the need arises.